File: polyglot/app_yaml_static_config/py/src/app_yaml_static_config/sdk.py
from typing import Any, Dict, List, Optional
import json
import glob
import os
from .core import AppYamlConfig
from .types import InitOptions

class AppYamlConfigSDK:
    def __init__(self, config: AppYamlConfig):
        self.config = config

    @classmethod
    def from_directory(cls, config_dir: str) -> 'AppYamlConfigSDK':
        # Simple implementation assuming .yaml files in dir
        files = glob.glob(os.path.join(config_dir, "*.yaml"))
        AppYamlConfig.initialize(InitOptions(files=files, config_dir=config_dir))
        return cls(AppYamlConfig.get_instance())

    def get(self, key: str) -> Any:
        return json.loads(json.dumps(self.config.get(key)))

    def get_nested(self, keys: List[str]) -> Any:
         return json.loads(json.dumps(self.config.get_nested(*keys)))

    def get_all(self) -> Dict[str, Any]:
        return json.loads(json.dumps(self.config.get_all()))

    def list_providers(self) -> List[str]:
        providers = self.config.get('providers', {})
        return list(providers.keys())

    def list_services(self) -> List[str]:
         services = self.config.get('services', {})
         return list(services.keys())
    
    def list_storages(self) -> List[str]:
        storages = self.config.get('storages', {})
        return list(storages.keys())

File: polyglot/app_yaml_static_config/mjs/src/sdk.ts
import { glob } from 'glob';
import * as path from 'path';
import { AppYamlConfig } from './core.js';
import { InitOptions } from './types.js';

export class AppYamlConfigSDK {
    private config: AppYamlConfig;

    constructor(config: AppYamlConfig) {
        this.config = config;
    }

    static async fromDirectory(configDir: string): Promise<AppYamlConfigSDK> {
        const files = await glob(path.join(configDir, "*.yaml"));
        await AppYamlConfig.initialize({ files, configDir });
        return new AppYamlConfigSDK(AppYamlConfig.getInstance());
    }

    get(key: string): unknown {
        const value = this.config.get(key);
        return value === undefined ? undefined : JSON.parse(JSON.stringify(value));
    }

    getNested(keys: string[]): unknown {
        const value = this.config.getNested(keys);
        return value === undefined ? undefined : JSON.parse(JSON.stringify(value));
    }

    getAll(): Record<string, unknown> {
        return JSON.parse(JSON.stringify(this.config.getAll()));
    }

    listProviders(): string[] {
        const providers = this.config.get<Record<string, unknown>>('providers') ?? {};
        return Object.keys(providers);
    }

    listServices(): string[] {
        const services = this.config.get<Record<string, unknown>>('services') ?? {};
        return Object.keys(services);
    }

    listStorages(): string[] {
        const storages = this.config.get<Record<string, unknown>>('storages') ?? {};
        return Object.keys(storages);
    }
}
